src/openrct2/management/NewsItem.cpp

```diff
@@ -52,10 +50,23 @@ bool news_item_is_valid_idx(int32_t index)
 }
 
 NewsItem* news_item_get(int32_t index)
+{
+    return gNewsItems.At(index);
+}
+
+NewsItem& NewsItemQueue::operator[](size_t index)
+{
+    if (index < MAX_RECENT_NEWS_ITEMS)
+        return Recent[index];
+    else
+        return Old[index - MAX_RECENT_NEWS_ITEMS];
+}
+
+NewsItem* NewsItemQueue::At(int32_t index)
```

Comment:
Looks like this could be const

Reply:
It could but it shouldn't, because it returns a non-const access to an internal member.
I could create a const version of this method, if you'd like.

Reply:
The queue itself is immutable. Let's take the example of shared ptr: https://en.cppreference.com/w/cpp/memory/shared_ptr/get

Returns a non-const pointer from an immutable store, thus the method is marked const.

Reply:
This is with respect to bitwise constness vs logical constness (see chapter 3 in Effective C++ by Scott Meyers). 

> What does it mean for a member function to be const? There are two
prevailing notions: bitwise constness (also known as physical constness) and logical constness.
> ...
> Surely there is something wrong when you create a constant object with a particular value and you invoke only const member functions on it, yet you still change its value!
> This leads to the notion of logical constness. Adherents to this philosophy argue that a const member function might modify some of the bits in the object on which it's invoked, but only in ways that clients cannot detect.

The reason the committee chose to make shared_ptr::get() const is simply because if it wasn't the API would be much harder to use. The 'get' method already is a "let me do dangerous things" method, so there is no point in protecting it.

If you take STL containers (for example [std::vector](https://en.cppreference.com/w/cpp/container/vector/operator_at)) you will see all of them overload operator at with non-const and const versions, return non-const or const references respectively.

Is it ok if I add a const version of the method?

Reply:
> Is it ok if I add a const version of the method?

Yes, that's fine.

I initially checked vector as well, but discarded it as here we deal explicitly with pointers to items, so I went to first thing that returns pointers I could think of. I do like your argumentation and let's do the overload then.